.github/workflows/docker-publish-opr-node-images.yaml

```diff
@@ -0,0 +1,84 @@
+name: Push Docker images to GHCR with Caching
+
+on:
+  workflow_dispatch:
+    inputs:
+      commit_sha:
+        description: 'Specific Commit SHA (Required)'
+        required: true
+      release_tag:
+        description: 'Release Tag (Optional)'
+        required: false
+        default: ''
+
+env:
+  REGISTRY: ghcr.io
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+    - name: Checkout repository at specified commit
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ github.event.inputs.commit_sha }}
+
+    - name: Setup Buildx
+      uses: docker/setup-buildx-action@v1
+      with:
+        install: true
+        driver-opts: image=moby/buildkit:master
+
+    - name: Cache Docker layers
+      uses: actions/cache@v2
+      with:
+        path: /tmp/.buildx-cache
+        key: ${{ runner.os }}-buildx-${{ github.event.inputs.commit_sha }}
+        restore-keys: |
+          ${{ runner.os }}-buildx-
+      if: ${{ success() }}
+
+    - name: Log into registry ${{ env.REGISTRY }}
+      uses: docker/login-action@v2
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+      if: ${{ success() }}
+
+    - name: Set Tag Name
+      id: set_tag
+      run: echo "::set-output name=tag::${{ github.event.inputs.release_tag || github.event.inputs.commit_sha }}"
```

Comment:
`::set-output` is [deprecated](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/#examples)